test/ServiceManagerTest.php

```diff
@@ -364,4 +365,51 @@ public function testResolvedAliasNoMatchingAbstractFactoryReturnsFalse()
 
         self::assertFalse($serviceManager->has('Alias'));
     }
+
+    /**
+     * Hotfix #3
+     * @see https://github.com/laminas/laminas-servicemanager/issues/3
+     */
+    public function testConfigureMultipleTimesAvoidsDuplicates()
+    {
+        $delegatorFactory = function (
+            ContainerInterface $container,
+            $name,
+            callable $callback
+        ) {
+            /** @var InvokableObject $instance */
+            $instance = $callback();
+            $options = $instance->getOptions();
+            $inc = $options['inc'] ?? 0;
+            return new InvokableObject(['inc' => ++$inc]);
+        };
+
+        $config = [
+            'factories' => [
+                'Foo' => function () {
+                    return new InvokableObject();
+                },
+            ],
+            'delegators' => [
+                'Foo' => [
+                    $delegatorFactory,
+                    LazyServiceFactory::class,
+                ],
+            ],
+            'lazy_services' => [
+                'class_map' => [
+                    'Foo' => InvokableObject::class,
+                ],
+            ],
+        ];
+
+        $serviceManager = new ServiceManager($config);
+        $serviceManager->configure($config);
+
+        /** @var InvokableObject $instance */
+        $instance = $serviceManager->get('Foo');
+
+        self::assertInstanceOf(InvokableObject::class, $instance);
```

Comment:
This assertion is not needed (based on `/** @var */` right above)

Reply:
The `@var` is only an IDE hint and not actually checking that the instance returned is the correct class tho, right?

Reply:
It will likely be picked up by static analysis later, but we can ignore it for now :+1: